docs/config.rst

```diff
@@ -1,11 +1,505 @@
+This allows one task in a flow to reference the `return_value` set on another task that executed prior to it.
+
+To reference a return value on a previous task use the following::
+
+    ^^prior_task.return_value
+
+
+In order to know what is available for ``<return_value>`` we need to find the source code for an individual task.
+Let's examine the definition for the ``upload_beta`` task. The internal ``cumulusci.yml`` file defines it as follows:
+
+.. code-block:: yaml
+
+    upload_beta:
+            description: Uploads a beta release of the metadata currently in the packaging org
+            class_path: cumulusci.tasks.salesforce.PackageUpload
+            group: Release Operations
+
+This informs us that we need to find where the class ``cumulusci.tasks.salesforce.PackageUpload`` is defined to see if anything is being set on ``self.return_values``.
+Some digging yields that this class is defined in the file `package_upload.py <>`_ and has a method called ``_set_return_values()``.
+`This method <https://github.com/SFDO-Tooling/CumulusCI/blob/3cad07ac1cecf438aaf087cdeff7b781a1fc74a1/cumulusci/tasks/salesforce/package_upload.py#L165>`_ sets ``self.return_values`` to a dictionary with the following keys: ``verison_number``, ``version_id``, and ``package_id``.
+
+Let's now look at the the ``release_beta`` flow as its defined in the internal cumulusci.yml file:
+
+.. code-block:: yaml
+
+   release_beta:
+        description: Upload and release a beta version of the metadata currently in packaging
+        steps:
+            1:
+                task: upload_beta
+                options:
+                    name: Automated beta release
+            2:
+                task: github_release
+                options:
+                    version: ^^upload_beta.version_number
+            3:
+                task: github_release_notes
+                ignore_failure: True  # Attempt to generate release notes but don't fail build
+                options:
+                    link_pr: True
+                    publish: True
+                    tag: ^^github_release.tag_name
+                    include_empty: True
+                    version_id: ^^upload_beta.version_id
+            4:
+                task: github_master_to_feature
+
+This flow references both ``version_id`` and ``version_number`` return values set on the ``upload_beta`` task.
+
+
+
+Using Tasks and Flows From a Different Project
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+With CumulusCI, it's also possible to use arbitrary tasks and flows from another project.
+To do this, the other project must be named in the ``sources`` section of the ``cumulusci.yml``:
+
+.. code-block:: yaml
+
+    sources:
+      npsp:
+        github: https://github.com/SalesforceFoundation/NPSP
+
+This says that when tasks or flows are referenced using the `npsp` namespace, CumulusCI should fetch the source from this GitHub repository.
```

Comment:
We should probably include a note that in order for this to work, the referenced repository needs to be readable. i.e. either it's public, or CumulusCI's github service is configured with the token of a user who has read access to it